viamonvif/service.go

```diff
@@ -136,11 +136,20 @@ func (dis *rtspDiscovery) Close(_ context.Context) error {
 	return nil
 }
 
-func createCamerasFromURLs(l CameraInfo, logger logging.Logger) ([]resource.Config, error) {
+func createCamerasFromURLs(l CameraInfo, discoveryDependencyName string, logger logging.Logger) ([]resource.Config, error) {
```

Comment:
This threading of the discovery dependency and which urls/configs are gaining a discovery dependency for config generation feels a bit clumsy. The root of why* it mostly has to look this way is because CameraInfo is used for [json serialization](https://github.com/viam-modules/viamrtsp/blob/0f7334dfc7a544aaa9038fc32cbd44b985adfa7f/viamonvif/discovery.go?plain=1#L107). And I didn't feel comfortable proposing a change to that user facing API.

Reply:
would it be worth making this function a method on the discovery service instead? then we could 

- check the mapped devices directly
- grab the name from the service rather than passing it in

prob makes it alot harder to test and this works as is, so probably not worth the effort.

Reply:
> would it be worth making this function a method on the discovery service instead? then we could

Agreed, that's a reasonable direction.

> prob makes it alot harder to test and this works as is, so probably not worth the effort.

I'll take your offer to leave it as-is.